dev-ruby/google-protobuf/google-protobuf-3.23.3.ebuild

```diff
@@ -0,0 +1,51 @@
+# Copyright 2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+USE_RUBY="ruby30 ruby31 ruby32"
+
+RUBY_FAKEGEM_EXTRADOC="README.md"
+
+RUBY_FAKEGEM_GEMSPEC="${PN}.gemspec"
+
+RUBY_FAKEGEM_EXTENSIONS=(ext/google/protobuf_c/extconf.rb)
+RUBY_FAKEGEM_EXTENSION_LIBDIR=lib/google
+
+inherit ruby-fakegem
+
+PARENT_PN="${PN/google-/}"
+PARENT_PV="$(ver_cut 2-)"
+PARENT_P="${PARENT_PN}-${PARENT_PV}"
+
+DESCRIPTION="Protocol Buffers are Google's data interchange format"
+HOMEPAGE="https://developers.google.com/protocol-buffers"
+SRC_URI="https://github.com/protocolbuffers/protobuf/archive/v${PARENT_PV}.tar.gz -> ${PARENT_P}.tar.gz"
+RUBY_S="${PARENT_P}/ruby"
+
+LICENSE="BSD"
+SLOT=$(ver_cut 1)
+KEYWORDS="~amd64"
+IUSE=""
+
+BDEPEND="
+	test? (
+		dev-libs/protobuf:0/23.3.0
```

Comment:
dev-libs/protobuf is interesting because the version went from 3.19.x:0/30.x (with libprotobuf-30.x.so) to 21.x:0/32.x (libprotobuf.32.x.so), and now 22.x onwards is 22.x:0/22.x.0? and 23.x:0/23.x.0? (libprotobuf.23.x.so). Beyond that, packages like dev-ruby/google-protobuf still follows the 3.23.x versioning.  I don't think that the slot number for 22.x onwards is yet well-defined which makes prereqs just as confusing. I grabbed the slot for protobuf from https://bugs.gentoo.org/867943 but it isn't something set in stone.

Reply:
And I don't think that it RDEPENDs on it because it is compiled and put into the package:
`/usr/lib64/ruby/gems/3.1.0/extensions/x86_64-linux/3.1.0/google-protobuf-3.23.3/google/protobuf_c.so`